package.json

```diff
@@ -27,7 +28,9 @@
     "@types/on-finished": "^2.3.1",
     "chalk": "^2.4.2",
     "on-finished": "^2.3.0",
-    "path-to-regexp": "^3.0.0",
-    "prom-client": "^12.0.0"
+    "path-to-regexp": "^3.0.0"
+  },
+  "peerDependencies": {
+    "prom-client": ">=11 <=12"
```

Comment:
I need to get my head around these one day.
How does it know that version 11 will work, when built against the version 12 dev dependency?

Reply:
https://github.com/LOKE/http-kit/actions/runs/4528841013

So I've added an new github action (called build) that does a `npm ci` then `npm i -D prom-client@{version}` and tries to build the app.  When it does a publish though it uses the version of the dev dependencies which should be up to date (in theory)

Reply:
<img width="518" alt="Screenshot 2023-03-27 at 4 28 37 pm" src="https://user-images.githubusercontent.com/7439294/227848847-3e78cd25-5400-4464-8bba-187e2ed17d97.png">

Reply:
Can we not make it support <=14?

Reply:
> How does it know that version 11 will work, when built against the version 12 dev dependency

It is up to the author to find a way to "know". The package JSON is just the author communicating the known supported versions.

In our case we use the GH actions to automatically check, but it could be down to checking through the list of breaking changes in new versions, similar to when we approve dependabot things.